core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/controller/promisehandlers/PythonDebugCommandHandler.scala

```diff
@@ -0,0 +1,21 @@
+package edu.uci.ics.amber.engine.architecture.controller.promisehandlers
+
+import com.twitter.util.Future
+import edu.uci.ics.amber.engine.architecture.controller.ControllerAsyncRPCHandlerInitializer
+import edu.uci.ics.amber.engine.architecture.controller.promisehandlers.PythonDebugCommandHandler.PythonDebugCommand
+import edu.uci.ics.amber.engine.architecture.pythonworker.promisehandlers.DebugCommandHandler.DebugCommand
+import edu.uci.ics.amber.engine.common.rpc.AsyncRPCServer.ControlCommand
+import edu.uci.ics.amber.engine.common.virtualidentity.ActorVirtualIdentity
+
+object PythonDebugCommandHandler {
+  final case class PythonDebugCommand(workerId: String, cmd: String) extends ControlCommand[Unit]
+}
+
+trait PythonDebugCommandHandler {
+  this: ControllerAsyncRPCHandlerInitializer =>
+  registerHandler { (msg: PythonDebugCommand, sender) =>
+    {
+      Future(send(DebugCommand(msg.cmd), ActorVirtualIdentity(msg.workerId))).unit
```

Comment:
this doesn't look right, `send()` already returns a future, and this line wraps a future inside another future, I believe the outer future will be resolved instantly, maybe just `send(DebugCommand(msg.cmd), ActorVirtualIdentity(msg.workerId))` is what you need, and it will compile

Reply:
right this outer Future is redundant. I will remove it.